mautic/mautic
# Plugin field mapping / priority of update flow / Salesforce sync push command

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR changes how fields are mapped in plugins. Firstly it will show the fields you have saved on your plugins already, if you want to add another field mapping with your integration then you can do so by clicking on add field. you will be able to select what integration field will be mapped with a mautic field and also indicate the direction of the priority of the field. so when you sync it will say how fields should be overwritten.
[//]: # ( As applicable: )
#### Steps to test this PR:
1. First check that fields on all your previously configured plugins are still showing.
2. Select fields and check to see if they are updated in Mautic and if they are to be pushed to an integration then check to see that only the fields with an arrow pointing to the integration fields are updated in your integration with Mautic fields.
3. If you have a salesforce integration check that a new feature is added to push leads to integration and the previous one was renamed to triggered action (this one continues to work the same, as it will push a contact through an action in a form or campaign). 
4. To test the sync command select this new feature, save and run this command on CLI: 
`php app/console mautic:integration:fetchleads --time-interval="29 days" --env=dev --integration=Salesforce` it should run the command where it will either update or create leads/contacts in Salesforce
